src/uniq.js

```diff
@@ -18,4 +20,39 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = _curry1(function uniq(list) {
+  /* global Set */
+  var item, set, idx, len, items, uniqs;
+
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  if (typeof Set === 'function') {
```

Comment:
Yup, that all makes sense. Would there be any objection to me updating the build script to emit a warning if it detects "detached" top-level variables/functions?